cloudslang-content-maven-compiler/src/main/java/io/cloudslang/maven/compiler/CloudSlangMavenCompiler.java

```diff
@@ -0,0 +1,148 @@
+package io.cloudslang.maven.compiler;
+
+import io.cloudslang.lang.compiler.SlangCompiler;
+import io.cloudslang.lang.compiler.SlangCompilerImpl;
+import io.cloudslang.lang.compiler.SlangSource;
+import io.cloudslang.lang.compiler.configuration.SlangCompilerSpringConfig;
+import io.cloudslang.lang.compiler.modeller.result.ExecutableModellingResult;
+import org.codehaus.plexus.compiler.AbstractCompiler;
+import org.codehaus.plexus.compiler.CompilerConfiguration;
+import org.codehaus.plexus.compiler.CompilerException;
+import org.codehaus.plexus.compiler.CompilerMessage;
+import org.codehaus.plexus.compiler.CompilerOutputStyle;
+import org.codehaus.plexus.compiler.CompilerResult;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.util.DirectoryScanner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by hanael on 10/07/2016.
+ */
+
+@Component(role = org.codehaus.plexus.compiler.Compiler.class , hint = "cloudslang" )
+public class CloudSlangMavenCompiler extends AbstractCompiler{
+
+    private SlangCompiler slangCompiler = new SlangCompilerImpl();
```

Comment:
is it necessary to create here with `new` - the fields wouldn't be populated from it anyway (Spring)